### PR TITLE
[Android][Feature][58897] Profile screen + Watching history + Add to my list

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Feature 58897: Profile screen](https://edu-redmine.sun-asterisk.vn/issues/xxxx)
## WHAT
-  Profile screen, login, logout
- Add to my list
- Save watching history, continue watching
## Evidence (Screenshot or Video)

Screenshot_20230116-161421_Movie-Streaming

Screenshot_20230116-161428_Movie-Streaming

Screenshot_20230116-161446_Movie-Streaming

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -43,4 +53,5 @@
             </intent-filter>
         </activity>
     </application>
-</manifest>
+
+</manifest>
```
Comment (hungld-1302) [inline app/src/main/AndroidManifest.xml]: endline

File: app/src/main/java/vn/ztech/software/movie_streaming/utils/extensions/IntExtension.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package vn.ztech.software.movie_streaming.utils.extensions
+
+fun Int.toBrightness(): Int{
+    return this * 255 / 100
+}
+
+fun Int.toPercentageString(): String{
+    return "Zoom: $this%"
+}
```
Comment (minhnn-1279) [inline app/src/main/java/vn/ztech/software/movie_streaming/utils/extensions/IntExtension.kt]: endline nhé a :v

**Review by hungvv-1301 — APPROVED** [review]

**Review by hungld-1302 — APPROVED** [review]

**Review by minhnn-1279 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: app/src/main/java/vn/ztech/software/movie_streaming/ui/player/MediaPlayerActivity.kt [hunk 3]
```diff
@@ -166,6 +265,17 @@ class MediaPlayerActivity<T : Media>() :
 
         viewModel.loading.observe(this) {
```
Comment (daolq-2712) [inline app/src/main/java/vn/ztech/software/movie_streaming/ui/player/MediaPlayerActivity.kt:284]: ?
Không handle gì à em?

File: app/src/main/java/vn/ztech/software/movie_streaming/ui/player/MediaPlayerActivity.kt [hunk 4]
```diff
@@ -138,6 +218,25 @@ class MediaPlayerActivity<T : Media>() :
             }
         }
 
+        viewModel.mediaDetailsWatchingHistory.observe(this){
+            if (it==null) return@observe
+            if(viewModel.mediaDetails.value != null){
+                viewModel.streamingStates.position = it.getEpisodeWatchingHistory(viewModel.mediaDetails.value?.selectedEpisode)?.position?:0L
+            }else{
```
Comment (daolq-2712) [inline app/src/main/java/vn/ztech/software/movie_streaming/ui/player/MediaPlayerActivity.kt]: .
format code

**Review by daolq-2712 — CHANGES_REQUESTED** [review]
Pls check my comments!

**Review by daolq-2712 — APPROVED** [review]

**Pull request merged**